odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,331 @@
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+try:
+    from NiftyPy.NiftyRec import (SPECT_project_parallelholes,
+                                  SPECT_backproject_parallelholes)
+    NIFTYREC_AVAILABLE = True
+except ImportError:
+    NIFTYREC_AVAILABLE = False
+
+import numpy as np
+
+from odl.operator import Operator
+from odl import uniform_discr_frompartition
+from odl.discr.lp_discr import DiscreteLp
+from odl.tomo.geometry.spect import ParallelHoleCollimatorGeometry
+
+
+__all__ = ('AttenuatedRayTransform', 'AttenuatedRayBackprojection',
+           'NIFTYREC_AVAILABLE')
+_SUPPORTED_IMPL = ('niftyrec', 'niftyrec_gpu')
+
+
+class AttenuatedRayTransform(Operator):
+
+    """The discrete attenuated Ray transform between L^p spaces."""
+
+    def __init__(self, dom, geometry, attenuation=None, psf=None,
+                 impl='niftyrec'):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Discretized space, a 3 dimensional domain of the forward projector
+        geometry : `ParallelHoleCollimatorGeometry`
+            The geometry of the transform. An ODL geometry instance
+            that contains information about the operator range
+        attenuation : domain `element-like` structure of the shape
+            `dom.shape`, optional
+            Linear attenuation map for SPECT.
+        psf : `element-like` structure, optional
+            Point Spread Functions for the correction of the
+            collimator-detector response
+        impl : {'niftyrec', 'niftyrec_gpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+
+            'niftyrec' : `NiftyRec`_ tomography toolbox using the `NiftyPy`_
+            Python wrapper and CPU
+
+            'niftyrec_gpu' : `NiftyRec`_ tomography toolbox using the
+            `NiftyPy`_ Python wrapper and GPU
+
+        References
+        ----------
+        .. _NiftyRec: http://niftyrec.scienceontheweb.net/wordpress/
+
+        .. _NiftyPy: https://github.com/spedemon/niftypy
+
+        """
+        if not isinstance(dom, DiscreteLp):
+            raise TypeError('discretized domain {!r} is not a `DiscreteLp`'
+                            ' instance.'.format(dom))
+
+        if not isinstance(geometry, ParallelHoleCollimatorGeometry):
+            raise TypeError('geometry {!r} is not a'
+                            '`ParallelHoleCollimatorGeometry` instance.'
+                            ''.format(geometry))
+
+        ran = uniform_discr_frompartition(geometry.partition)
+        super().__init__(dom, ran, linear=True)
+
+        impl, impl_in = str(impl).lower(), impl
+        if impl not in _SUPPORTED_IMPL:
+            raise ValueError('implementation {!r} not supported.'
+                             ''.format(impl_in))
+
+        if impl.startswith('niftyrec'):
+            if not NIFTYREC_AVAILABLE:
```

Comment:
Yes it is, the only thing here is that in the current version of NiftyRec,  `niftyrec_gpu` performs consistently only with volume size `128 x 128 x 128`.